pkgs/servers/mautrix-telegram/default.nix

```diff
@@ -1,95 +1,73 @@
-{ lib, python3, mautrix-telegram, fetchFromGitHub
+{ lib
+, python3
+, mautrix-telegram
+, fetchFromGitHub
 , withE2BE ? true
+, withHQthumbnails ? false
 }:
 
 let
   python = python3.override {
     packageOverrides = self: super: {
-      asyncpg = super.asyncpg.overridePythonAttrs (oldAttrs: rec {
-        version = "0.25.0";
-        src = oldAttrs.src.override {
-          inherit version;
-          hash = "sha256-Y/jmppczsoVJfChVRko03mV/LMzSWurutQcYcuk4JUA=";
-        };
-      });
-      mautrix = super.mautrix.overridePythonAttrs (oldAttrs: rec {
-        version = "0.16.3";
-        src = oldAttrs.src.override {
-          inherit (oldAttrs) pname;
-          inherit version;
-          sha256 = "sha256-OpHLh5pCzGooQ5yxAa0+85m/szAafV+l+OfipQcfLtU=";
-        };
-      });
       tulir-telethon = self.telethon.overridePythonAttrs (oldAttrs: rec {
-        version = "1.25.0a7";
+        version = "1.25.0a20";
         pname = "tulir-telethon";
         src = oldAttrs.src.override {
           inherit pname version;
-          sha256 = "sha256-+wHRrBluM0ejdHjIvSk28wOIfCfIyibBcmwG/ksbiac=";
+          sha256 = "sha256-X9oo+YCNMqQrJvQa/PIi9dFgaeQxbrlnwUJnwjRb6Jc=";
         };
       });
     };
   };
-
-  # officially supported database drivers
-  dbDrivers = with python.pkgs; [
-    psycopg2
-    aiosqlite
-    # sqlite driver is already shipped with python by default
-  ];
-
 in python.pkgs.buildPythonPackage rec {
   pname = "mautrix-telegram";
-  version = "0.11.3";
+  version = "0.12.0";
   disabled = python.pythonOlder "3.8";
 
   src = fetchFromGitHub {
     owner = "mautrix";
     repo = "telegram";
     rev = "v${version}";
-    sha256 = "sha256-PfER/wqJ607w0xVrFZadzmxYyj72O10c2lIvCW7LT8Y=";
+    sha256 = "sha256-SUwiRrTY8NgOGQ643prsm3ZklOlwX/59m/u1aewFuik=";
   };
 
   patches = [ ./0001-Re-add-entrypoint.patch ];
 
   propagatedBuildInputs = with python.pkgs; ([
-    Mako
-    aiohttp
-    mautrix
-    sqlalchemy
-    CommonMark
     ruamel-yaml
     python-magic
+    CommonMark
+    aiohttp
+    yarl
+    mautrix
     tulir-telethon
-    telethon-session-sqlalchemy
+    asyncpg
+    Mako
+    # optional
+    cryptg
+    cchardet
+    aiodns
+    brotli
     pillow
-    lxml
-    setuptools
+    qrcode
+    phonenumbers
     prometheus-client
+    aiosqlite
+  ] ++ lib.optionals withHQthumbnails [
+    moviepy
   ] ++ lib.optionals withE2BE [
-    asyncpg
     python-olm
     pycryptodome
     unpaddedbase64
-  ]) ++ dbDrivers;
+  ]);
 
-  # Tests are broken and throw the following for every test:
-  #   TypeError: 'Mock' object is not subscriptable
-  #
-  # The tests were touched the last time in 2019 and upstream CI doesn't even build
-  # those, so it's safe to assume that this part of the software is abandoned.
```

Comment:
Does this comment still apply? If so I would like to at least add back that the tests are not functional/broken.

Reply:
The tests just do not exists now, they were completely removed.